src/auto-evo/simulation/SimulationCache.cs

```diff
@@ -20,28 +20,28 @@
 /// </remarks>
 public class SimulationCache
 {
+    public readonly Dictionary<(MicrobeSpecies, SelectionPressure), float> CachedPressureScores = new();
```

Comment:
This should be private.

Reply:
It is referenced from the cachedScore. Not sure if it is possible to make private while cachedScore still exists. Well unless I moved that to this class, which in hindsight probably makes more sense.

Reply:
I did try in my IDE making this private and it didn't give an error, but it may not have seen that reference immediate. Better to move all the caching stuff here instead of exposing something that no one outside this class should be allowed to modify for code encapsulation reasons.